### PR TITLE
Enable race detector for ci builds

### PR DESCRIPTION
**Pull request by tylerwilliams**

### CODE REVIEW

**Review by siggisim — APPROVED** [review]

**Review by vadimberezniker — APPROVED** [review]

**Review by bduffany — APPROVED** [review]

File: .bazelrc [hunk 1]
```diff
@@ -53,6 +53,7 @@ build:ci --repository_cache=/home/runner/repo-cache/
 build:ci --flaky_test_attempts=2
 build:ci --color=yes
 build:ci --disk_cache=
+build:ci --@io_bazel_rules_go//go/config:race
```
Comment (bduffany) [inline .bazelrc:56]: mind setting it for `build:workflows` too while you're here?

Comment (tylerwilliams) [reply .bazelrc:56]: Done.

**Comment by tylerwilliams** [comment]
(going to send another cl to fix all the tests before submitting this)

**Comment by tylerwilliams** [comment]
Will re-evaluate once https://github.com/buildbuddy-io/buildbuddy/pull/2062 is in

**Pull request merged**